src/resolve/resolved-mdns.c

```diff
@@ -262,12 +256,19 @@ int manager_mdns_ipv4_fd(Manager *m) {
                 goto fail;
         }
 
+        /* try to bind without SO_REUSEADDR option to detect another mDNS responder */
         r = bind(m->mdns_ipv4_fd, &sa.sa, sizeof(sa.in));
         if (r < 0) {
                 r = -errno;
                 goto fail;
         }
 
+        r = setsockopt(m->mdns_ipv4_fd, SOL_SOCKET, SO_REUSEADDR, &one, sizeof(one));
```

Comment:
> SO_REUSEADDR has an effect during bind, afaiu. Setting it afterward has no effect.

I am not sure about that. In socket(7), it is stated that the another option `SO_REUSEPORT` must be set before calling `bind`. However, for `SO_REUSEADDR`, there are no statements like that. Avahi actually sets `SO_REUSEADDR` after `bind`.